pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryImport.xml

```diff
@@ -1474,8 +1474,281 @@ public class Foo {
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.io.File;
+public class Foo {
+    void test() {
+        list.stream().map(File::getName);
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in switch expressions -->
+    <test-code>
+        <description>Import used in switch expression</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import some.pkg.MyEnum;
+public class Foo {
+    String test(MyEnum e) {
+        return switch(e) {
+            case VALUE1 -> "one";
+            case VALUE2 -> "two";
+        };
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in records -->
+    <test-code>
+        <description>Import used in record component</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.time.LocalDate;
+public record Person(String name, LocalDate birthDate) {}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in sealed classes -->
+    <test-code>
+        <description>Import used in sealed class hierarchy</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import some.pkg.Shape;
+public sealed class Circle extends Shape permits BigCircle, SmallCircle {}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in pattern matching -->
+    <test-code>
+        <description>Import used in instanceof pattern matching</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.util.List;
+public class Foo {
+    void test(Object o) {
+        if (o instanceof List l) {
+            l.size();
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in text blocks -->
+    <test-code>
+        <description>Import used in text block with embedded expressions</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.time.LocalDate;
+public class Foo {
+    String test(String name) {
+        return """
+            Hello %s!
+            Today is %s
+            """.formatted(name, LocalDate.now());
+    }
 }
         ]]></code>
     </test-code>
 
+    <!-- Test cases for imports used in var declarations -->
+    <test-code>
+        <description>Import used in var declaration with initializer</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.util.ArrayList;
+public class Foo {
+    void test() {
+        var list = new ArrayList<String>();
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in module-info.java -->
+<!--    <test-code>-->
+<!--        <description>Import used in module-info requires</description>-->
+<!--        <expected-problems>0</expected-problems>-->
+<!--        <code><![CDATA[-->
+<!--import some.module.ModuleName;-->
```

Comment:
This doesn't sound correct, you cannot import modules that way. What you can do is, to import on demand all packages that a module exports via `import module ...` (https://openjdk.org/jeps/476). But you wouldn't use that in a module-info.java file...